full-stack-engineer/submissions/michaelyao/client/src/components/addReview/AddReview.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import {
+  Dialog,
+  Box,
+  Typography,
+  IconButton,
+  Grid,
+  Rating,
+  TextField,
+  DialogActions,
+  Button,
+  Alert,
+  Snackbar,
+} from '@mui/material';
+import CancelIcon from '@mui/icons-material/Cancel';
+import { createReview } from '../../Api';
+
+const AddReview = ({ productId, onHideAddReview }) => {
+  const [ratingValue, setRatingValue] = useState(0);
+  const [content, setContent] = useState('');
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [displaySuccessAlert, setDisplaySuccessAlert] = useState(false);
+  const [reviewCreated, setReviewCreated] = useState(false);
+
+  const handleClose = () => {
+    onHideAddReview(reviewCreated);
```

Comment:
I got it :D 
Here I will suggest for simplicity that you have two different handlers.  `onReviewCreated` and `onClose` (which is not created). Then you can save a flag and from the parent is also more clear what is every handler for.

Reply:
Yep. I updated. Hope I understand it correctly.